packages/sdk-codegen/src/python.gen.ts

```diff
@@ -263,9 +278,28 @@ ${this.hooks.join('\n')}
   }
 
   initArg(indent: string, property: IProperty) {
-    return `${indent}self.${property.name} = ${property.name}`
+    let propName = property.name
+    if (this.pythonKeywords.has(propName)) {
+      propName = propName + '_'
+    }
+    return `${indent}self.${propName} = ${propName}`
+  }
+
+  typeProperties(type: IType) {
+    return Object.values(type.requiredProperties).concat(Object.values(type.optionalProperties))
   }
 
+  private isBareForwardRef = (prop: IProperty) => {
+    // contains a "bare" forward reference e.g. `result_format: "ResultFormat"`
+    // so we need to emit the `__annotations__` property
+    return (
+      prop.required &&
+      (prop.type.customType || prop.type.className == 'EnumType') &&
+      !['ArrayType', 'HashType', 'DelimArrayType'].includes(
+        prop.type.className
+      )
```

Comment:
and that this further filters the result to a "singular"/"bare" model reference. e.g.

`role: "Role"`

Not `role: Sequence["Role"]` or `role: MutableMapping[str, "Role]`